pkg/docker/config/config.go

```diff
@@ -541,6 +542,28 @@ func getPathToAuthWithOS(sys *types.SystemContext, goOS string) (string, bool, e
 // or returns an empty dockerConfigFile data structure if auth.json does not exist
 // if the file exists and is empty, readJSONFile returns an error
 func readJSONFile(path string, legacyFormat bool) (dockerConfigFile, error) {
+	if _, err := os.Stat(path); err != nil {
+		if os.IsNotExist(err) {
+			var auths dockerConfigFile
+			auths.AuthConfigs = map[string]dockerAuthConfig{}
+			return auths, nil
+		}
+		return dockerConfigFile{}, err
+	}
+
+	// Make sure to read-lock the file before reading it.
+	lock, err := lockfile.GetLockfile(path)
```

Comment:
Isn’t the path parameter a path of the _lock_? Which contains a random value to support `.Modified`? How can `auth.json` be a JSON and a random string at the same time?

Reply:
`Touch()` and `Modified()` don't work in this scenario but I think that's OK in this scenario since we're controlling it here.

Reply:
Ah, I have not been reading that carefully enough; the contents are not actually _used_ if we call neither `Touch()` nor `Modified()`.

Still, that feels like an implementation detail that could change any time. I suppose our tests would catch it…